test/ParseNumber.cpp

```diff
@@ -0,0 +1,932 @@
+  EXPECT_EQ(-1, i32);
+  EXPECT_TRUE(ParseNumber("-0x80000000", &i32));
+  EXPECT_EQ(std::numeric_limits<int32_t>::min(), i32);
+}
+
+TEST(ParseUnsignedIntegers, Sample) {
+  uint32_t u32;
+
+  // Invalid parse.
+  EXPECT_FALSE(ParseNumber("", &u32));
+  EXPECT_FALSE(ParseNumber("0=", &u32));
+
+  // Valid values.
+  EXPECT_TRUE(ParseNumber("0", &u32));
+  EXPECT_EQ(0u, u32);
+  EXPECT_TRUE(ParseNumber("4294967295", &u32));
+  EXPECT_EQ(std::numeric_limits<uint32_t>::max(), u32);
+  EXPECT_FALSE(ParseNumber("4294967296", &u32));
+
+  // Hex values.
+  EXPECT_TRUE(ParseNumber("0xffffffff", &u32));
+  EXPECT_EQ(std::numeric_limits<uint32_t>::max(), u32);
+
+  // We don't care about -0 since it's rejected at a higher level.
+  EXPECT_FALSE(ParseNumber("-1", &u32));
+}
+
+TEST(ParseWideSignedIntegers, Sample) {
+  int64_t i64;
+  EXPECT_FALSE(ParseNumber("", &i64));
+  EXPECT_FALSE(ParseNumber("0=", &i64));
+  EXPECT_TRUE(ParseNumber("0", &i64));
+  EXPECT_EQ(0, i64);
+  EXPECT_TRUE(ParseNumber("0x7fffffffffffffff", &i64));
+  EXPECT_EQ(0x7fffffffffffffff, i64);
+  EXPECT_TRUE(ParseNumber("-0", &i64));
+  EXPECT_EQ(0, i64);
+  EXPECT_TRUE(ParseNumber("-1", &i64));
+  EXPECT_EQ(-1, i64);
+}
+
+TEST(ParseWideUnsignedIntegers, Sample) {
+  uint64_t u64;
+  EXPECT_FALSE(ParseNumber("", &u64));
+  EXPECT_FALSE(ParseNumber("0=", &u64));
+  EXPECT_TRUE(ParseNumber("0", &u64));
+  EXPECT_EQ(0u, u64);
+  EXPECT_TRUE(ParseNumber("0xffffffffffffffff", &u64));
+  EXPECT_EQ(0xffffffffffffffffULL, u64);
+
+  // We don't care about -0 since it's rejected at a higher level.
+  EXPECT_FALSE(ParseNumber("-1", &u64));
+}
+
+TEST(ParseFloat, Sample) {
+  float f;
+
+  EXPECT_FALSE(ParseNumber("", &f));
+  EXPECT_FALSE(ParseNumber("0=", &f));
+
+  // These values are exactly representatble.
+  EXPECT_TRUE(ParseNumber("0", &f));
+  EXPECT_EQ(0.0f, f);
+  EXPECT_TRUE(ParseNumber("42", &f));
+  EXPECT_EQ(42.0f, f);
+  EXPECT_TRUE(ParseNumber("2.5", &f));
+  EXPECT_EQ(2.5f, f);
+  EXPECT_TRUE(ParseNumber("-32.5", &f));
+  EXPECT_EQ(-32.5f, f);
+  EXPECT_TRUE(ParseNumber("1e38", &f));
+  EXPECT_EQ(1e38f, f);
+  EXPECT_TRUE(ParseNumber("-1e38", &f));
+  EXPECT_EQ(-1e38f, f);
+}
+
+TEST(AssemblyContextParseFloat, Overflow) {
+  // The assembler parses using HexFloat<FloatProxy<float>>.  Make
+  // sure that succeeds for in-range values, and fails for out of
+  // range values.  When it does overflow, the value is set to the
+  // nearest finite value, matching C++11 behavior for operator>>
+  // on floating point.
+  spvutils::HexFloat<spvutils::FloatProxy<float>> f(0.0f);
+
+  EXPECT_TRUE(ParseNumber("1e38", &f));
+  EXPECT_EQ(1e38f, f.value().getAsFloat());
+  EXPECT_TRUE(ParseNumber("-1e38", &f));
+  EXPECT_EQ(-1e38f, f.value().getAsFloat());
+  EXPECT_FALSE(ParseNumber("1e40", &f));
+  EXPECT_FALSE(ParseNumber("-1e40", &f));
+  EXPECT_FALSE(ParseNumber("1e400", &f));
+  EXPECT_FALSE(ParseNumber("-1e400", &f));
+}
+
+TEST(AssemblyContextParseDouble, Sample) {
```

Comment:
Should remove `AssemblyContext` prefix in the test name.

Same for the following.